src/DropboxOauthButton.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState } from 'react';
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import CloudUploadIcon from '@material-ui/icons/CloudUpload';
+import { withStyles } from '@material-ui/core/styles';
+
+const styles = theme => ({
+    button: {
+        margin: theme.spacing.unit,
+    },
+    leftIcon: {
+        marginRight: theme.spacing.unit,
+    },
+    rightIcon: {
+        marginLeft: theme.spacing.unit,
+    },
+    iconSmall: {
+        fontSize: 20,
+    },
+});
+
+const DropboxOauthButton = ({ getToken, isAuthenticated, classes }) => {
+    const [dialogOpen, setDialogOpen] = useState(false);
+    const [inputText, setInputText] = useState(undefined);
+
+    const handleClickOpenDialog = () => {
+        setDialogOpen(true);
+    };
+
+    const handleClickCancel = () => {
+        setDialogOpen(false);
+        setInputText(undefined);
+    };
+
+    const handleClickDoAuth = () => {
+        setDialogOpen(false);
+        setInputText(undefined);
+        getToken && getToken(inputText);
+    };
+
+    const handleDialogClose = () => {
+        setDialogOpen(false);
+    };
```

Comment:
It might be simpler to have 
```javascript
const openDialog = setDialogOpen(true);
const closeDialog = setDialogOpen(false);

const resetDialog = () => {
  closeDialog();
  setInputText(null);
}

const onDoAuth = ()  {
  getToken && getToken(inputText);
  resetDialog();
}
```
and use those in the component props.

Note that the logic is same. Except, I am trying to make it easier to see whats happening. E.g. you can read the logic.